components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/IdentityProviderDataRetrievalClient.java

```diff
@@ -102,6 +112,131 @@ public String getIdPImage(String tenant, String idpName)
+                return configMap;
+            }
+            JSONObject idpDetailedResult = executePath(tenant, idpURLWithId);
+            if (idpDetailedResult == null) {
+                return configMap;
+            }
+            JSONObject federatedAuthObject = idpDetailedResult.getJSONObject(FEDERATED_AUTHENTICATORS);
+            if (federatedAuthObject == null) {
+                return configMap;
+            }
+            JSONArray federatedAuthenticators = federatedAuthObject.getJSONArray(AUTHENTICATORS);
+            if (federatedAuthenticators == null) {
+                return configMap;
+            }
+            String federatedIDPURLWithID = null;
+            for (int i = 0; i < federatedAuthenticators.length(); i++) {
+                JSONObject property = (JSONObject) federatedAuthenticators.get(i);
+                if (idpCode.equals(property.getString(NAME))) {
+                    federatedIDPURLWithID = property.getString(SELF_LINK);
+                    break;
+                }
+            }
+            if (StringUtils.isEmpty(federatedIDPURLWithID)) {
+                return configMap;
+            }
+
+            // i.e. /t/carbon.super/api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc/
+            // federated-authenticators/R29vZ2xlT0lEQ0F1dGhlbnRpY2F0b3I
+            JSONObject federatedIdpResult = executePath(tenant, federatedIDPURLWithID);
+            if (federatedIdpResult == null) {
+                return configMap;
+            }
+            JSONArray federatedIdpProperties = federatedIdpResult.getJSONArray(PROPERTIES);
+            if (federatedIdpProperties.length() == 0) {
+                return configMap;
+            }
+            for (int i = 0; i < federatedIdpProperties.length(); i++) {
+                JSONObject property = (JSONObject) federatedIdpProperties.get(i);
+                if (configKeys.contains(property.getString(KEY))) {
+                    configMap.putIfAbsent(property.getString(KEY), property.getString(VALUE));
+                }
+            }
+        } catch (JSONException ex) {
+            throw new IdentityProviderDataRetrievalClientException(
+                    "Error while decoding the JSON object for federated IDP configs", ex);
+        }
+        return configMap;
+    }
+
+    /**
+     * This function executes the HTTP GET calls of given path on carbon API and returns the resulted JSON object
+     *
+     * @param tenant String. The tenant name of the session i.e. carbon.super.
+     * @param path   String. The context or to call without host details. i.e.
+     *               api/server/v1/identity-providers?filter=name+eq+Google
+     *               api/server/v1/identity-providers/6719c5cc-5162-44c7-9190-cb74d500f5fc
+     * @return JSONObject. The resulted JSONObject of the GET call
+     * @throws IdentityProviderDataRetrievalClientException When there is an error with executing the get calls
```

Comment:
Let's end with `.` at the end of each statement.

Reply:
Fixed 38d0c51fd441a7ee513b0b52d2459c3a732d9e2d